blocks/show-hide-group/index.js

```diff
@@ -0,0 +1,80 @@
+import { registerBlockType } from '@wordpress/blocks';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { dispatch, useSelect } from '@wordpress/data';
+
+// Internal dependencies.
+import metadata from './block.json';
+import './style.css';
+
+// Register the block.
+registerBlockType(metadata, {
+	edit: (props) => {
+		const blockProps = useBlockProps(); // eslint-disable-line react-hooks/rules-of-hooks
+		const { setAttributes } = props;
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		const innerBlocks = useSelect((select) => {
+			const currentBlocks = select('core/block-editor').getBlocks(
+				props.clientId
+			);
+			return currentBlocks;
+		});
+
+		// Update inner blocks' htmlId attributes when the inner blocks change.
+		let newId = '';
+		let currentIds = '';
+		for (let i = 0; i < innerBlocks.length; i++) {
+			newId = 'show-hide-section-' + i;
```

Comment:
Multiple block support:
This is the kind of tricky part - I guess the easiest way forward here would be to prepend the `show-hide-group` block's client ID to this...